airqo-platform/AirQo-frontend
# Ft user management

- What does this PR do?
This PR showcases the implementation of user management tasks as highlighted below. I will use PR comments to walk you through the codebase since I am a proponent of using descriptive variable names rather than many comments in code. 😎 Most of the following implementations have corresponding backend implementations. 

1. Different views/components based on a user's access right. Admin can see collaborators they created. Super admin can see see all admins while collaborators cannot be able to see other users. Refactored a lot of code in the process of achieving this - took a lot of my time!
2. Dialog components for editing/updating, deleting, activating candidates that expressed interest and adding collaborators/users by admins.
3. Added a registration button on the landing page which redirects to the landing page after successful registration. This is accompanied by a success alert.
4. Automatic logout of a user when idle (idle timeout).
5. Updated/created the corresponding Redux files for all the components which were created. These included: `usersTable`, `UserEditForm` and `UsersToolbar`.
6. Created the `Defaults` dumb component and container- also created its corresponding redux implementations.
7. Adopted `material-ui `for some components that were using react-bootstrap - just to maintain consistency.

- What JIRA task is related to this PR?
The JIRA cards are here:

1. [Integrate joining platform with the project so that we’re not using external link](https://airqoteam.atlassian.net/browse/NLYTCS-35?atlOrigin=eyJpIjoiMzY2OTVkZDJiNmE4NDQwMGI0ODg3MDM2YjgwZTgzNWUiLCJwIjoiaiJ9)
2. [Administrator able to set up user](https://airqoteam.atlassian.net/browse/NLYTCS-27?atlOrigin=eyJpIjoiY2YyMDIzY2ZiZWIxNGY2ZTkxZTdmODJkMzBlMTJjYzgiLCJwIjoiaiJ9)
3. [Activate functionality so KCCA can register their team users(user management)](https://airqoteam.atlassian.net/browse/NLYTCS-38?atlOrigin=eyJpIjoiZGNjZDI2YWU5ODU0NGEyYjkxMjI0NDg2ZjFlNzBjMWMiLCJwIjoiaiJ9)
4. [Settings: Each user needs to be able to save defaults easily](https://airqoteam.atlassian.net/browse/NLYTCS-36?atlOrigin=eyJpIjoiNmZkYmQwNzIzZTU2NGJiZjgzZjFjYmYxMTE0YjNiNmYiLCJwIjoiaiJ9)
5. [Ensuring users from different organisations have access to only the devices related to their organisations](https://airqoteam.atlassian.net/browse/NLYTCS-64?atlOrigin=eyJpIjoiNGRhNmRlNjkwOTQyNDI3Mzg3MTgxOGVkMzU1M2MzOGIiLCJwIjoiaiJ9)

- How do I test out this PR?
Git clone this repo `cd AirQo-api/src/analytics` Then run `npm install` to get all the necessary dependencies. The unit tests have not yet been carried out so this PR is not yet ready for testing.

- Which end points should are ready for testing?
There are no endpoints for testing this PR.
